### ISSUE TITLE
Design/Functionality for Weather Tile?

### ISSUE THREAD

**Issue by nelsonic** [numo-labs/lambda-ne-weather-provider#1]
@emiledskar can you please confirm the design (and functionality) for the weather tiles so we know what data we need to fetch from the yr.no API. (_thanks!_)

**Comment by nelsonic** [comment]
@emiledskar shared:
https://cdn.zeplin.io/57862d8c724b75ab5d685d18/screens/cache/681949746a2134c4fd176509d4fd2718e7692937.png
![weather design](https://cdn.zeplin.io/57862d8c724b75ab5d685d18/screens/cache/681949746a2134c4fd176509d4fd2718e7692937.png)

Are we displaying the weather forecast for the _next week_ or the date when the person is attempting to book their holiday for?

**Comment by emiledskar** [comment]
@nelsonic: The date when the person is attempting to book feels more sensible. I have to add that these are quick sketches, so feel free to change icons and rearrange the layout as you see fit.

**Comment by lennym** [comment]
Worth noting that it's highly unlikely that any weather forecast API will be able to provide forecasts for more than a week ahead, so a forecast for a proposed holiday date might not be possible.

**Comment by emiledskar** [comment]
True, let's go with a week ahead then.

**Comment by nelsonic** [comment]
That's why I was asking. Because long-range forecasts are unreliable and I don't see the _benefit_ of having a short-term forecast. Why does the traveller care what the weather in Mallorca is _today_ (_or in the next week_) when she is only travelling in 2 months time...? 😕

**Comment by lennym** [comment]
I'd suggest a combination of weather and climate data.

Show big data of what the weather is like _right now_, with some supporting information about typical temperatures and weather at time of travel.

Crudely mocked up thusly:

![untitled-1](https://cloud.githubusercontent.com/assets/117398/17737948/3a86438c-6487-11e6-9a43-caf89acc2bf0.png)

**Comment by emiledskar** [comment]
@nelsonic Yes 😕 I think it is more useful for the user to have some kind of general heat guide that reflects what temperatures one could expect in x area for each month. Is this info something we can get from yr.no API ?

**Comment by emiledskar** [comment]
Maybe something like this then. To show what the typical temperature is for the same month as been selected for the search. And then in the detail view we can show info about other months and some other info like precipitation.
![image](https://cloud.githubusercontent.com/assets/8721221/17738995/d764580e-6493-11e6-8540-e2e8adc489e1.png)

**Comment by nelsonic** [comment]
@emiledskar / @lennym agree 100%, that's why I was asking... 😉 
Giving the **_average temperature**_ in the **desired travel period** is _way_ more **_useful**_ / **_relevant**_ in the decision process (IMO). 
The high-low bar-chart is good as a _detailed_/_expanded_ view but for a simple card the average temp is simpler. 

It _appears_ that spies.dk already _has_ the average temperature info:
![spies-dk-hotel-temperature-table](https://cloud.githubusercontent.com/assets/194400/17740751/9c97d25c-6491-11e6-8ef0-c605ead6f546.png)

http://www.spies.dk/vaelg-hotellet?ItemId=83203 scroll and click on "Vejr og klima"

@jruts asked Jesper for access to the data but was told to source it from no.yr ... 
I think we are just not asking the "_right_" questions. (_or possibly people are too busy to help us...?_)
But the data is _definitely_ in the Spies.dk Database and we _should_ be able to get access to it.

**Comment by lennym** [comment]
> source it from no.yr

That would be a lot easier if the documentation wasn't all in Norwegian.

**Comment by emiledskar** [comment]
If you can get the same info as they are presenting at spies I think we should go with something like this for the details view (if we can get info about precipitation that would be nice to add as well).  I think we can leave out the information about the current weather as it does not really seem  to fit the use case.
![image](https://cloud.githubusercontent.com/assets/8721221/17767807/c5bdb0c0-6531-11e6-9cd5-5cd8b5efa36c.png)

And the tile could look like this.
![image](https://cloud.githubusercontent.com/assets/8721221/17767906/30e4bfe2-6532-11e6-8336-ad978106bed3.png)

**Comment by nelsonic** [comment]
@lennym yeah, the no.yr docs in Norwegian are challenge. 
But we _really_ should be re-purposing the data that Spies already.

@emiledskar your sketches look good. (_I would suggest simplifying the first iteration to get it done sooner then enhance in response to user feedback..._)
concerned that there are `29 hours` of sunshine per day (`soltimer/dag`) in July/August ... 😜

**Comment by emiledskar** [comment]
@nelsonic I agree on making it simple first and then enhance later. I think the most important insight we want now is if people will find the tile interesting. So I will give you free hands to choose the easiest approach to display some relevant data. Then we can change it later if we see that people are clicking in the tiles.
Oh yeah! The sun is shining so bright at Mallorca that the days gain some extra hours ;)

**Comment by lennym** [comment]
If that's the data that we want to display then yr.no is not going to work for us. It only has short term forecast data (that I can find anyway, the site being in Norwegian doesn't make it easy) and no long term climate data.

Also, doing lookups for a particular location is a mild pain - we'd need to hardcode the specific url for each location as they're quite particular on yr.no (e.g. http://www.yr.no/place/Spain/Catalonia/Barcelona/forecast.xml) and it doesn't obviously seem like it's possible to do latlng lookups.

Also also, they insist that if you're using their data then you display a message and link alongside it - see also http://www.yr.no/place/Spain/Catalonia/Barcelona/forecast.xml

tl;dr - we need somewhere better to get the data from.

**Comment by emiledskar** [comment]
Let's leave the design open until you know exactly what data we can get from yr or spies. Just let me know when you know and I will update the design to fit what data we have. 

Also, Roy asked me to remember you guys that the image shown on this tile should be different from the one shown on the destination tile :)

**Comment by lennym** [comment]
From yr.no we can get the weather for locations for the next week, with certain caveats - probably most pertinently the fact that we have to display the attribution message alongside the data, and they cap requests at one every 10 minutes.

From spies, I wouldn't know how to begin finding out that information (@jruts may know more than me)> Although if all we want is month-by-month average temperature/rainfall data then, in the worst case scenario, we can probably just scrape the data once since it's not going to be something that ever really changes much.

**Comment by emiledskar** [comment]
I like the last option you mentioned. As you said, the information we want is not going to change that much over time.
